### ISSUE TITLE
bazel_4 install check fails on aarch64-darwin

### ISSUE THREAD

**Issue by uri-canva** [NixOS/nixpkgs#145587]
### Steps To Reproduce
1. Enable `doInstallCheck` in `bazel_4`
2. Build on `aarch64-darwin`

### Expected behavior
Build succeeds.

### Additional context
https://github.com/NixOS/nixpkgs/issues/134758

### Notify maintainers
@mboes 

### Metadata
 - system: `"aarch64-darwin"`
 - host os: `Darwin 20.6.0, macOS 11.5.2`
 - multi-user?: `yes`
 - version: `nix-env (Nix) 2.3.15`
 - channels(root): `"nixpkgs-21.11.pre309670.253aecf69ed"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

**Comment by uri-canva** [comment]
Current error reported by @bundiyan in https://github.com/NixOS/nixpkgs/issues/134758#issuecomment-940384499:
```
ERROR: /private/tmp/nix-build-bazel-4.2.1.drv-0/bazel_L7PWK63b/out/external/googleapis/BUILD.bazel:201:14: Generating Java (Immutable) proto_library @googleapis//:google_devtools_build_v1_build_status_proto failed: (Segmentation fault): protoc failed: error executing command 
  (cd /private/tmp/nix-build-bazel-4.2.1.drv-0/bazel_L7PWK63b/out/execroot/io_bazel && \
  exec env - \
    PATH=/nix/store/14riyxn9qdjab9mpy6riw2wp97wgg5bl-zip-3.0/bin:/nix/store/nf5jsmymmnz9jxb7s3x0rq40nhf6ffnb-python3-3.9.6/bin:/nix/store/ac212rlx03j6iyi9vzwfhrhx037jc552-unzip-6.0/bin:/nix/store/8y8kcppkv356zk18kkapz2358g9vgnlk-which-2.21/bin:/nix/store/hk8bzk1z63a77f1zydbrdj808hmr9ipa-bash/bin:/nix/store/4lm9am37hnyx3h99w5xzwrsbk91hmh45-cctools-port-949.0.1/bin:/nix/store/ywnnx39dgd515cj4fv9z97mmga8qcbwd-clang-wrapper-11.1.0/bin:/nix/store/wb2g1gnnjwzpcw43pahdmahybzryvhr0-clang-11.1.0/bin:/nix/store/98ggz6lvray5v2kym17yylnnyfz4f4lr-coreutils-8.32/bin:/nix/store/sy8c8580ym2ry7b8ihmggz84a2p16ham-cctools-binutils-darwin-wrapper-949.0.1/bin:/nix/store/7lfjbip2dvhvnhvdj0kx4d63apqlz83v-cctools-binutils-darwin-949.0.1/bin:/nix/store/5sbk39gzizr0cb3ziclhslii6zv71js0-zulu11.48.21-ca-jdk-11.0.11/bin:/nix/store/98ggz6lvray5v2kym17yylnnyfz4f4lr-coreutils-8.32/bin:/nix/store/l1s52qqk0hngyqfjv4938mnc0sbgsjqw-findutils-4.8.0/bin:/nix/store/1ba2zyd9gyal97na97r073g7v6vi87zx-diffutils-3.7/bin:/nix/store/xy0hjszca945iq44z6s29x4n1cc27347-gnused-4.8/bin:/nix/store/fv9ljbpicbqb8h0a40qnz5fmk9s3kb8l-gnugrep-3.6/bin:/nix/store/ijq2s2cn18834ybzsinj1638s2pwp07m-gawk-5.1.0/bin:/nix/store/z4nx7dqji7ziw2hk3apadfc72xjszxww-gnutar-1.34/bin:/nix/store/sqkc7rkjql1r159x903f4r05m2j1sgvn-gzip-1.10/bin:/nix/store/pyr76kmnl8frc71xiyw24q9y7s13jilh-bzip2-1.0.6.0.2-bin/bin:/nix/store/57dwqr6igadm896armq61h96psdni160-gnumake-4.3/bin:/nix/store/cm0ch90wmxl2llj0xjn4rdvmgjn8nr3q-bash-4.4-p23/bin:/nix/store/kki238b0g3l7h5jzndq3l91mpr3b44sc-patch-2.7.6/bin:/nix/store/gx6k0r2zdgfc3mva6fhn7ngdmxyxs5r8-xz-5.2.5-bin/bin:/nix/store/cm0ch90wmxl2llj0xjn4rdvmgjn8nr3q-bash-4.4-p23/bin:/nix/store/98ggz6lvray5v2kym17yylnnyfz4f4lr-coreutils-8.32/bin:/nix/store/l1s52qqk0hngyqfjv4938mnc0sbgsjqw-findutils-4.8.0/bin:/nix/store/ijq2s2cn18834ybzsinj1638s2pwp07m-gawk-5.1.0/bin:/nix/store/fv9ljbpicbqb8h0a40qnz5fmk9s3kb8l-gnugrep-3.6/bin:/nix/store/3zlmz891mnc15flsjrn2wfi97kikicg1-gnutar-1.34/bin:/nix/store/xy0hjszca945iq44z6s29x4n1cc27347-gnused-4.8/bin:/nix/store/sqkc7rkjql1r159x903f4r05m2j1sgvn-gzip-1.10/bin:/nix/store/8y8kcppkv356zk18kkapz2358g9vgnlk-which-2.21/bin:/nix/store/ac212rlx03j6iyi9vzwfhrhx037jc552-unzip-6.0/bin:/nix/store/bnlk9hc1bry2kypj3kc1bz2pxv4h6w88-file-5.40/bin:/nix/store/14riyxn9qdjab9mpy6riw2wp97wgg5bl-zip-3.0/bin:/nix/store/rs534hr95wf5y6wyhdhsg7rd7wdr6zcf-python-2.7.18/bin:/nix/store/nf5jsmymmnz9jxb7s3x0rq40nhf6ffnb-python3-3.9.6/bin \
  bazel-out/darwin_arm64-opt-exec-EDC14992/bin/external/com_google_protobuf/protoc '--proto_path=external/googleapis' '--proto_path=bazel-out/darwin_arm64-opt/bin/external/com_google_protobuf/_virtual_imports/descriptor_proto' '--proto_path=bazel-out/darwin_arm64-opt/bin/external/com_google_protobuf/_virtual_imports/any_proto' '--java_out=bazel-out/darwin_arm64-opt/bin/external/googleapis/google_devtools_build_v1_build_status_proto-speed-src.jar' '-Igoogle/devtools/build/v1/build_status.proto=external/googleapis/google/devtools/build/v1/build_status.proto' '-Igoogle/api/annotations.proto=external/googleapis/google/api/annotations.proto' '-Igoogle/api/http.proto=external/googleapis/google/api/http.proto' '-Igoogle/protobuf/descriptor.proto=bazel-out/darwin_arm64-opt/bin/external/com_google_protobuf/_virtual_imports/descriptor_proto/google/protobuf/descriptor.proto' '-Igoogle/protobuf/any.proto=bazel-out/darwin_arm64-opt/bin/external/com_google_protobuf/_virtual_imports/any_proto/google/protobuf/any.proto' --direct_dependencies google/api/annotations.proto:google/protobuf/any.proto:google/devtools/build/v1/build_status.proto '--direct_dependencies_violation_msg=%s is imported, but @googleapis//:google_devtools_build_v1_build_status_proto doesn'\''t directly depend on a proto_library that '\''srcs'\'' it.' external/googleapis/google/devtools/build/v1/build_status.proto)
Execution platform: //:default_host_platform
```

**Comment by jacobfoard** [comment]
The install check seems like it should work now with https://github.com/NixOS/nixpkgs/pull/151503 being merged in, but the cpp test is failing now, maybe some nix interaction with the fix that was put in for https://github.com/bazelbuild/bazel/issues/14273? As if if I use the latest `bazel_4` the test binary gets killed, but if I just build `//examples/cpp:hello-world` and try to run the resulting binary it is fine. 

However using the release bazel bin, it is able to compile and run the cpp test.

I wasn't able to track down what interaction there is between the tests and the normal executable, but happy to help debug if anyone has insight into that.

**Comment by wegank** [comment]
Closed by #147649.
